langchain/vectorstores/pgvector.py

```diff
@@ -291,8 +291,15 @@ def similarity_search_with_score_by_vector(
         if filter is not None:
             filter_clauses = []
             for key, value in filter.items():
-                filter_by_metadata = EmbeddingStore.cmetadata[key].astext == str(value)
-                filter_clauses.append(filter_by_metadata)
+                IN = "IN"
+                if isinstance(value, str):
+                    filter_by_metadata = EmbeddingStore.cmetadata[
+                        key].astext == str(value)
+                    filter_clauses.append(filter_by_metadata)
+                elif isinstance(value, dict) and IN in value:
+                    filter_by_metadata = EmbeddingStore.cmetadata[
+                        key].astext.in_(value[IN])
+                    filter_clauses.append(filter_by_metadata)
```

Comment:
I think there's a bug in the code w/ respect to handling of `int` / `float` values. 

In the previous version: 

{ 'page': 5 }

Would've resulted in a filter being issued with the number `5` being converted to the string literal "5".

In the new version, no filter clause will be generated.

To fix the behavior, could you put the `dict` check as the first `if statement`, and use the previous behavior for the `else` branch that will use the str conversion logic for everything that's not a dict?

Reply:
done